### ISSUE TITLE
Doesn't work with engine

### ISSUE THREAD

**Issue by gossi** [chancancode/ember-concurrency-async#1]
Looks like this doesn't work with engines. Here is the short error snippet:

```
Build Error (Analyzer)
Duplicate plugin/preset detected.
If you'd like to use two separate instances of a plugin,
they need separate names, e.g.
  plugins: [
    ['some-plugin', {}],
    ['some-plugin', {}, 'some unique name'],
  ]
```

Looks like some broccoli plugin is registered twice ?!? Maybe this error is already enough as a hint to point to the problem. If not please ping I can work on a repro case for more detailed errors.

**Comment by chancancode** [comment]
Hmm, I thought I guarded against this case here: https://github.com/chancancode/ember-concurrency-async/blob/master/index.js#L12-L13, not sure why this is still happening.

**Comment by teddyzeenny** [comment]
`hasPlugin` and `addPlugin` are not looking at the same plugin name. So `hasPlugin` is always `false`, rendering the guard useless. I did some investigation 🕵️ , and it seems that:
- Given the `require.resolve`-ed path of the plugin module as `Plugin`.
- `addPlugin` takes `Plugin` and registers it with the name `ember-concurrency-async` (same as the full package's name). [This is the code path](https://github.com/dfreeman/ember-cli-babel-plugin-helpers/blob/4c8db6df3a86270008b0bf43762dfbedf3f54fd3/src/-private/plugin-names.ts#L9) that resolves the `Plugin` path into this name.
- `hasPlugin` takes `Plugin` but instead of resolving it to `ember-concurrency-async`, it [calls normalize on it](https://github.com/dfreeman/ember-cli-babel-plugin-helpers/blob/a7d437d835eff36c1152c77cd91f66af05a22ffb/src/-private/plugin-configuration.ts#L15), which [keeps it exactly as it is](https://github.com/dfreeman/ember-cli-babel-plugin-helpers/blob/4c8db6df3a86270008b0bf43762dfbedf3f54fd3/src/-private/plugin-names.ts#L33), a full path to the module. 
- When the comparison happens, we compare `'ember-concurrency-async' === '/path/to/module/ember-concurrency-async/lib/babel-plugin-transform-ember-concurrency-async-tasks'` which will result in a false negative.

I'm not familiar enough with how `ember-cli-babel-plugin-helpers` works to figure out the best solution. My instinct was that we should call resolve instead of normalize on `Plugin` inside `hasPlugin`. This would probably resolve the issue but we'd need to change how `hasPlugin` works and I don't know if this will break existing things. Or maybe the addon should not be saved using the package's name inside `addPlugin`? Maybe someone has better ideas on how to solve this here or in `ember-cli-babel-plugin-helpers`.

**Comment by jamescdavis** [comment]
@gossi, @teddyzeenny #11 updates the minimum `ember-cli-babel-helpers` to a version that fixes this issue.

**Comment by jamescdavis** [comment]
Released as [`ember-concurrency-async@0.3.2`](https://www.npmjs.com/package/ember-concurrency-async/v/0.3.2).
